dpnp/dpnp_iface_mathematical.py

```diff
@@ -2742,6 +2777,180 @@ def gradient(f, *varargs, axis=None, edge_order=1):
+    --------
+    The x-coordinate sequence is expected to be increasing, but this is not
+    explicitly enforced.  However, if the sequence `xp` is non-increasing,
+    interpolation results are meaningless.
+
+    Note that, since NaN is unsortable, `xp` also cannot contain NaNs.
+
+    A simple check for `xp` being strictly increasing is::
+
+        import dpnp as np
+        np.all(np.diff(xp) > 0)
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> xp = np.array([1, 2, 3])
+    >>> fp = np.array([3 ,2 ,0])
+    >>> x = np.array([2.5])
+    >>> np.interp(x, xp, fp)
+    array([1.])
+    >>> x = np.array([0, 1, 1.5, 2.72, 3.14])
+    >>> np.interp(x, xp, fp)
+    array([3.  , 3.  , 2.5 , 0.56, 0.  ])
+    >>> x = np.array([3.14])
+    >>> UNDEF = -99.0
+    >>> np.interp(x, xp, fp, right=UNDEF)
+    array([-99.])
+
+    Interpolation with periodic x-coordinates:
+
+    >>> x = np.array([-180, -170, -185, 185, -10, -5, 0, 365])
+    >>> xp = np.array([190, -190, 350, -350])
+    >>> fp = np.array([5, 10, 3, 4])
+    >>> np.interp(x, xp, fp, period=360)
+    array([7.5 , 5.  , 8.75, 6.25, 3.  , 3.25, 3.5 , 3.75])
+
+    Complex interpolation:
+
+    >>> x = np.array([1.5, 4.0])
+    >>> xp = np.array([2,3,5])
+    >>> fp = np.array([1.0j, 0, 2+3j])
+    >>> np.interp(x, xp, fp)
+    array([0.+1.j , 1.+1.5j])
+
+    """
+
+    dpnp.check_supported_arrays_type(x, xp, fp)
+
+    if xp.ndim != 1 or fp.ndim != 1:
+        raise ValueError("xp and fp must be 1D arrays")
+    if xp.size != fp.size:
+        raise ValueError("fp and xp are not of the same length")
+    if xp.size == 0:
+        raise ValueError("array of sample points is empty")
+
+    usm_type, exec_q = get_usm_allocations([x, xp, fp])
+
+    x_dtype = dpnp.common_type(x, xp)
+    x_float_type = dpnp.default_float_type(exec_q)
+
+    if not dpnp.can_cast(x_dtype, x_float_type):
+        raise TypeError(
+            "Cannot cast array data from"
+            f" {x_dtype} to {x_float_type} "
+            "according to the rule 'safe'"
+        )
+
+    x = dpnp.asarray(x, dtype=x_float_type, order="C")
+    xp = dpnp.asarray(xp, dtype=x_float_type, order="C")
+
+    out_dtype = dpnp.common_type(x, xp, fp)
+
+    fp = dpnp.asarray(fp, dtype=out_dtype, order="C")
+
+    if period is not None:
+        period = _validate_interp_param(period, "period", exec_q, usm_type)
+        if period == 0:
+            raise ValueError("period must be a non-zero value")
+        period = dpnp.abs(period)
+
+        # left/right are ignored when period is specified
+        left = None
+        right = None
+
+        # normalizing periodic boundaries
+        x %= period
+        xp %= period
+        asort_xp = dpnp.argsort(xp)
+        xp = xp[asort_xp]
+        fp = fp[asort_xp]
+        xp = dpnp.concatenate((xp[-1:] - period, xp, xp[0:1] + period))
+        fp = dpnp.concatenate((fp[-1:], fp, fp[0:1]))
+        assert xp.flags.c_contiguous
+        assert fp.flags.c_contiguous
```

Comment:
What is the intention of that assertions?

Reply:
To make sure that `xp` and `fp` are contiguous after normalizing periodic boundaries

Reply:
But isn't `dpnp.concatenate` return a new array always? We never used the assertation to ensure contiguous of the data before passing to the extension and I wonder why it's needed here.